src/api/Services/Geocoder/LocationRequester.cs

```diff
@@ -61,16 +61,23 @@ public async Task<int> RequestLocations()
             var failures = new Dictionary<string, string>();
             foreach (var location in locations)
             {
-
-                var coordinates = await geocoder.ResolvePostCodeAsync(location.GeoAddress);
-                if (coordinates != null)
+                // The try catch needs to be here as geocoder consume external service
+                try
                 {
-                    location.FormattedAddress = coordinates.FormattedLocation;
-                    location.Longitude = coordinates.Longitude;
-                    location.Latitude = coordinates.Latitude;
-                    location.LastGeocodedUtc = utcNow;
+                    var coordinates = await geocoder.ResolvePostCodeAsync(location.GeoAddress);
+                    if (coordinates != null)
+                    {
+                        location.FormattedAddress = coordinates.FormattedLocation;
+                        location.Longitude = coordinates.Longitude;
+                        location.Latitude = coordinates.Latitude;
+                        location.LastGeocodedUtc = utcNow;
+                    }
+                    else
+                    {
+                        failures.Add(location.Id.ToString(), location.GeoAddress);
+                    }
                 }
-                else
+                catch (Exception)
                 {
                     failures.Add(location.Id.ToString(), location.GeoAddress);
```

Comment:
can we include the exception info somewhere useful? maybe log to insights as a warning

Reply:
Good call, forgot about it.

Logged as error as if there is exception it is quite terminal.